compiler/rustc_lint/src/types.rs

```diff
@@ -742,12 +742,12 @@ fn lint_wide_pointer<'tcx>(
                 AmbiguousWidePointerComparisonsAddrSuggestion::Cast {
                     deref_left,
                     deref_right,
-                    // those two Options are required for correctness as having
-                    // an empty span and an empty suggestion is not permitted
-                    left_before: (l_ty_refs != 0).then_some(left),
-                    right_before: (r_ty_refs != 0).then(|| r_span.shrink_to_lo()),
-                    left: l_span.shrink_to_hi(),
-                    right,
+                    paren_left: (l_ty_refs != 0).then_some(")").unwrap_or_default(),
+                    paren_right: (r_ty_refs != 0).then_some(")").unwrap_or_default(),
```

Comment:
Cute, but even longer than the obvious "if" :grin: 
```suggestion
                    paren_left: if l_ty_refs != 0 { ")" } else { "" },
                    paren_right: if r_ty_refs != 0 { ")" } else { "" },
```